managedtenants/core/addons_loader/addon.py

```diff
@@ -250,23 +250,28 @@ def load_imageset(self, imageset_version):
                 f" {self.imagesets_path}"
             )
 
+        valid_imagesets = self.load_valid_imageset()
+        imageset = self.get_target_imageset(imagesets_iter=valid_imagesets)
+        self._validate_schema_instance(imageset, "imageset")
+        return imageset
```

Comment:
this is still returning one single imageset

Reply:
The plan is to use `load_valid_imagesets` and `get_target_imageset` function in the managed-tenants.
The `load_valid_imageset` will return me all the valid imageset for a particular addon meanwhile I can pass in the required version and insert that in ocm.

The code will somewhat look like this in `tasks/deploy/32_ocm_addons_service_upsert.py`:

```
addon = Addon(path=self.addon.metadata['id'],environment=self.environment.name)

valid_imagesets = addon.load_valid_imageset()

# this will give me all the imageset in the 
imageset =  os.listdir("./addons/<addon-id>/addonimagesets/production")

# perform some cleanup in imageset so that we get the versions, e.g. `rhoads.v1.1.1.yaml` changes to `1.1.1`
# this list will be named versions and will be sorted according to the semver.
# iterating through the list we will get the targeted version of imageset and that will be upserted in AS.

for version in versions:
     target_imageset = addon.get_target_imageset(valid_imagesets,target_version=version)
     ocm_cli.addons_service_upsert_addon_version(
           imageset = target_imageset,
           metadata = self.addon.metadata,
     )
```

Reply:
Why the unnecessary changes ?  We just have to write a new method which does the following:
```
def imagesets(self):
  imageset_yamls = map(self.load_yaml, self.get_available_imagesets())
  valid_imagesets = filter(is_not_none, imageset_yamls)
  return valid_imagesets
```

Reply:
made the changes.